app/controllers/forumthreads_controller.rb

```diff
@@ -80,6 +85,20 @@ def destroy
     redirect_to @thread.forum
   end
 
+  def search
+  end
+
+  def search_redirect
```

Comment:
When you submit the search form, it includes empty parameters. For example, if you use basic search and submit with a blank search box, the url will contain `?query=`. Is there a way to avoid that within the form other than using jQuery?